lib/msal-node/src/client/ClientCredentialClient.ts

```diff
@@ -58,7 +59,22 @@ export class ClientCredentialClient extends BaseClient {
 
         const cachedAuthenticationResult =
             await this.getCachedAuthenticationResult(request);
+
         if (cachedAuthenticationResult) {
+            // if the token is not expired but must be refreshed; get a new one in the background
+            if (this.lastCacheOutcome === CacheOutcome.REFRESH_CACHED_ACCESS_TOKEN) {
+                this.logger.info(
+                    "ClientCredentialClient:getCachedAuthenticationResult - Cached access token's refreshOn property has been exceeded'. It's not expired, but must be refreshed."
+                );
+
+                // return the cached token, don't wait for the result of this request
+                const accessTokenRefresh = true;
+                this.executeTokenRequest(request, this.authority, accessTokenRefresh);
+            }
+
+            this.lastCacheOutcome = "";
```

Comment:
> this.lastCacheOutcome = "";

there is an empty string constant somewhere if I'm not mistaken

Reply:
There is: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/6f5a21499f915fd5a49515c4d0927bf4be81fd07/lib/msal-common/src/utils/Constants.ts#L46